File: lib/models/user_model.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'package:json_annotation/json_annotation.dart';

part 'user_model.g.dart';

@JsonSerializable()
class UserModel {
  String? id;
  String? idCard;
  String? firstname;
  String? lastname;
  String? mobile;
  String? email;
  String? rank;
  String? affiliation;
  String? status;

  UserModel({
    this.id,
    this.idCard,
    this.firstname,
    this.lastname,
    this.mobile,
    this.email,
    this.rank,
    this.affiliation,
    this.status,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) =>
      _$UserModelFromJson(json);
  Map<String, dynamic> toJson() => _$UserModelToJson(this);
}

File: server/internal/api/v1/user/model/user.model.go
package model

type CreateUserDTO struct {
	IdCard    string `json:"idCard" validate:"required"`
	Firstname string `json:"firstname" validate:"required"`
	Lastname  string `json:"lastname" validate:"required"`
	Mobile    string `json:"mobile" validate:"required,min=10,max=10"`
	Email     string `json:"email" validate:"required"`
	// Image       string `json:"image" validate:"required"`
	Rank        string `json:"rank" validate:"required"`
	Affiliation string `json:"affiliation" validate:"required"`
	Status      string `json:"status" validate:"required"`
}
